terraform/modules/emblem-app/main.tf

```diff
@@ -78,3 +78,102 @@ resource "google_storage_bucket_iam_member" "sessions-iam" {
   role   = "roles/storage.objectAdmin"
   member = "serviceAccount:${google_service_account.website_manager.email}"
 }
+
+############################
+# E2E Delivery testing IAM #
+############################
+
+resource "google_project_iam_member" "delivery_gae_viewer_iam_member" {
+  project  = var.project_id
+  provider = google
+  role     = "roles/appengine.appViewer"
+  member   = "serviceAccount:${data.google_project.ops.number}@cloudbuild.gserviceaccount.com"
+
+  depends_on = [
+    google_project_service.emblem_app_services
+  ]
+}
+
+resource "google_project_iam_member" "delivery_sa_admin_iam_member" {
+  project  = var.project_id
+  provider = google
+  role     = "roles/iam.serviceAccountAdmin"
+  member   = "serviceAccount:${data.google_project.ops.number}@cloudbuild.gserviceaccount.com"
+
+  depends_on = [
+    google_project_service.emblem_app_services
+  ]
+}
+
+resource "google_project_iam_member" "delivery_compute_viewer_iam_member" {
+  project  = var.project_id
+  provider = google
+  role     = "roles/compute.viewer"
+  member   = "serviceAccount:${data.google_project.ops.number}@cloudbuild.gserviceaccount.com"
+
+  depends_on = [
+    google_project_service.emblem_app_services
+  ]
+}
+
+resource "google_project_iam_member" "delivery_iam_reviewer_iam_member" {
+  project  = var.project_id
+  provider = google
+  role     = "roles/iam.securityReviewer"
+  member   = "serviceAccount:${data.google_project.ops.number}@cloudbuild.gserviceaccount.com"
+
+  depends_on = [
+    google_project_service.emblem_app_services
+  ]
+}
+
+resource "google_project_iam_member" "delivery_storage_object_admin_granting_iam_member" {
+  project  = var.project_id
+  provider = google
+  role     = "roles/iam.securityAdmin"
+  member   = "serviceAccount:${data.google_project.target_project.number}@cloudbuild.gserviceaccount.com"
+
+  depends_on = [
+    google_project_service.emblem_ops_services
+  ]
+
+  condition {
+    title       = "Sessions Object Admin only"
+    description = "Only allow granting/revocation of the Storage Object Admin role to the Sessions bucket"
+    expression  = "api.getAttribute('iam.googleapis.com/modifiedGrantsByRole', []).hasOnly(['roles/storage.objectAdmin']) && resource.type == 'storage.googleapis.com/Bucket' && resource.name.endsWith('sessions')"
+  }
+}
+
+resource "google_project_iam_member" "delivery_run_admin_granting_iam_member" {
+  project  = var.project_id
+  provider = google
+  role     = "roles/iam.securityAdmin"
+  member   = "serviceAccount:${data.google_project.target_project.number}@cloudbuild.gserviceaccount.com"
+
+  depends_on = [
+    google_project_service.emblem_ops_services
+  ]
+
+  condition {
+    title       = "Cloud Run roles only"
+    description = "Only allow granting/revocation of roles relevant to Cloud Run"
+    expression  = "api.getAttribute('iam.googleapis.com/modifiedGrantsByRole', []).hasOnly(['roles/run.admin', 'roles/iam.serviceAccountUser', 'roles/datastore.user'])"
```

Comment:
This seems like it's combining roles to deploy a Cloud Run service with roles needed for API Cloud Run service to serve requests. Is this meant for Cloud Build permissions or Cloud Run permissions? If the latter, doesn't it also need object read/write for the website?

Reply:
These are the roles the (currently Cloud Build) service account needs to be able to _grant_ to other principals.

(The condition here is to prevent the service account's `Security Admin` permission from granting _arbitrary_ IAM roles.)

Reply:
I added comments here, and a cross-linking comment to `modules/emblem-app/iam.tf` 👍